Source/WebGPU/WGSL/Parser.cpp

```diff
@@ -840,10 +873,11 @@ Expected<UniqueRef<AST::Expression>, Error> Parser<Lexer>::parseUnaryExpression(
 {
     START_PARSE();
 
-    if (current().m_type == TokenType::Minus) {
+    if (canProceedSingularExpression(current())) {
+        auto op = toUnaryOperation(current());
         consume();
```

Comment:
Random thought, but maybe consume should also return `Token&`? That way you could write `toUnaryOperation(consume())`

Reply:
It would have to return a `Token` right? `consume()` bumps the internal storage of what the current token is.